lib/services/svg_renderer/index.js

```diff
@@ -0,0 +1,358 @@
+        // compute pageW using grid layout
+        options.pageW = options.pageMarginL + options.pageMarginR - options.itemPaddingX + (options.itemW + options.itemPaddingX) * options.itemsPerRow
+    // if page size is well defined
+    } else {
+        // then compute upper bounds for grid layout
+        const maxItemsPerRow  = Math.floor((effectivePageW - options.itemPaddingX) / (options.itemW + options.itemPaddingX))
+        const maxItemsPerCol  = Math.floor((effectivePageH - options.itemPaddingY) / (options.itemH + options.itemPaddingY))
+
+        // enforce upper bounds for grid layout
+        options.itemsPerRow  = Math.min(options.itemsPerRow , maxItemsPerRow ) || maxItemsPerRow
+        options.itemsPerCol  = Math.min(options.itemsPerCol , maxItemsPerCol ) || maxItemsPerCol
+
+        // compute maxItemsPerPage
+        const maxItemsPerPage = options.itemsPerRow * options.itemsPerCol
+        // enforce upper bound on maxItemsPerPage
+        options.itemsPerPage = Math.min(options.itemsPerPage, maxItemsPerPage)
+    }
+
+    options.numberOfPages = Math.ceil(options.numberOfItems / options.itemsPerPage)
+    for(let page = 0; page < options.numberOfPages; page ++)
+        options.pages.push(SVG().size(
+            options.pageW,
+            options.pageH,
+        ))
+}
+
+function _renderComponent (game, component, template, options) {
+    for(let item = 0; item < options.numberOfItems; item ++) {
+        let page = Math.floor( item / options.itemsPerPage )
+        let row  = Math.floor((item - options.itemsPerPage * page) / options.itemsPerRow      )
+        let col  =            (item - options.itemsPerPage * page) - options.itemsPerRow * row
+
+        options['renderContext'] = options.pages[page]
+        options['renderOffsetX'] = options.pageMarginL + (options.itemW + options.itemPaddingX) * col
+        options['renderOffsetY'] = options.pageMarginT + (options.itemH + options.itemPaddingY) * row
+        _renderItem(game, component, template, options.items[item], options)
+    }
+}
+
+function _prepareItem(game, component, template, item, options) {
+    if(!options.renderContext) {
+        // compute component size in pixels
+        options['itemW'] = convert(template.size.w, INCHES, PIXELS, options)
+        options['itemH'] = convert(template.size.h, INCHES, PIXELS, options)
+        options['renderContext'] = SVG().size(
+            options.itemW,
+            options.itemH
+        )
+        options['renderOffsetX'] = 0
+        options['renderOffsetY'] = 0
+    }
+}
+
+function _renderItem (game, component, template, item, options) {
+    const rootContext = options.renderContext
+    const itemContext = options.renderContext.nested()
+
+    itemContext.move(
+        options.renderOffsetX,
+        options.renderOffsetY
+    )
+    itemContext.size(
+        options.itemW,
+        options.itemH
+    )
+
+    itemContext.rect(100,100).fill('#f03').move(100,100)
```

Comment:
I rendered SOMETHING so i could tell when i had it working, toss this once you start rendering real stuff